projects/storefrontlib/src/cms-components/user/checkout-login/checkout-login.component.ts

```diff
@@ -1,30 +1,81 @@
-import { Component, OnInit } from '@angular/core';
-import { FormGroup, FormBuilder, Validators } from '@angular/forms';
-import { CartService } from '@spartacus/core';
+import { Component } from '@angular/core';
+import { AbstractControl, FormBuilder, Validators } from '@angular/forms';
+import { FormUtils } from '../../../shared/utils/forms/form-utils';
 import { CustomFormValidators } from '../../../shared/utils/validators/custom-form-validators';
+import {
+  CartService,
+  GlobalMessageService,
+  GlobalMessageType,
+} from '@spartacus/core';
 
 @Component({
   selector: 'cx-checkout-login',
   templateUrl: './checkout-login.component.html',
 })
-export class CheckoutLoginComponent implements OnInit {
-  form: FormGroup;
-
-  constructor(private fb: FormBuilder, private cartService: CartService) {}
-
-  ngOnInit(): void {
-    this.form = this.fb.group({
-      userId: ['', [Validators.required, CustomFormValidators.emailValidator]],
-      userIdConf: [
+export class CheckoutLoginComponent {
+  form = this.formBuilder.group(
+    {
+      email: ['', [Validators.required, CustomFormValidators.emailValidator]],
+      emailConfirmation: [
         '',
         [Validators.required, CustomFormValidators.emailValidator],
       ],
-      termsandconditions: [Validators.requiredTrue],
-    });
+      termsAndConditions: ['', [Validators.requiredTrue]],
+    },
+    { validator: this.emailsMatch }
+  );
+  private submitClicked = false;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    protected globalMessageService: GlobalMessageService,
+    protected cartService: CartService
+  ) {}
+
+  isNotValid(formControlName: string): boolean {
+    return FormUtils.isNotValidField(
+      this.form,
+      formControlName,
+      this.submitClicked
+    );
+  }
+
+  isEmailConfirmInvalid(): boolean {
+    return (
+      this.form.hasError('NotEqual') &&
+      (this.form.get('emailConfirmation').touched &&
+        this.form.get('emailConfirmation').dirty)
+    );
   }
 
-  submit(): void {
-    const email = this.form.value.userId;
+  onSubmit() {
+    this.submitClicked = true;
+
+    if (this.form.invalid) {
+      this.validateTermsAndConditions();
+      return;
+    }
+
+    const email = this.form.value.email;
     this.cartService.addEmail(email);
   }
+
+  private validateTermsAndConditions() {
+    const value = this.form.get('termsAndConditions').value;
+    if (value !== 'true') {
```

Comment:
Value is a boolean. It does not work with a string. It will display a message if the form is invalid even if the consent is checked.

Reply:
Did you test this? I tested it and it works with a string.